collector/perf_linux.go

```diff
@@ -15,16 +15,25 @@ package collector
 
 import (
 	"fmt"
-	"runtime"
-
 	perf "github.com/hodgesds/perf-utils"
 	"github.com/prometheus/client_golang/prometheus"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+	"runtime"
+	"strconv"
+	"strings"
 )
 
 const (
 	perfSubsystem = "perf"
 )
 
+var (
+	cpus                = kingpin.Flag("collector.perf.cpus", "List of CPUs from which perf metrics should be collected").Default("").String()
+	hwProfilerCpuMap    = make(map[*perf.HardwareProfiler]int)
+	swProfilerCpuMap    = make(map[*perf.SoftwareProfiler]int)
+	cacheProfilerCpuMap = make(map[*perf.CacheProfiler]int)
```

Comment:
These are global for all collectors, which could cause nasty conflicts/side effects. These variables need to be moved inside the `perfCollector` struct.

Reply:
'cpus' should also be renamed, for a global scope var it's to generic. perfCpus or perfCpusFlag - best to follow roughly what other collectors are doing.